app/controllers/api/v2/override_values_controller.rb

```diff
@@ -94,6 +97,17 @@ def return_if_override_mismatch
       def resource_class
         LookupValue
       end
+
+      def rename_use_puppet_default
+        return unless params[:override_value]
+
+        old_value = params[:override_value].delete(:use_puppet_default)
+
+        if old_value.present?
```

Comment:
`use_puppet_default` is a boolean and when it is false `old_value.present?` returns false and the value doesn't get changed.